### PR TITLE
5240 place2book returns error

### PR DESCRIPTION
**Pull request by arniloftsson**
#### Link to issue

https://platform.dandigbib.org/issues/5240

#### Description

This PR deals with a error that I have not been able to reproduce. The errors shows up as event_id and event_maker_id get set to 1 and the event loses connection to its counterpart in place2book service. This PR introduces som definsive programming and logging if the error occurs. It hinders the page from saving invalid event_id and event_maker_id to the database. And it changes to place moste likely to cause this errors to a more robust code. It also changes message text that would otherwise be confusing to users.

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md). 
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 87a2c0b16670a7bb5761bed79f92b8acc6bb6764
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 87a2c0b16670a7bb5761bed79f92b8acc6bb6764
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 87a2c0b16670a7bb5761bed79f92b8acc6bb6764
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

File: modules/ding_place2book/ding_place2book.fields.inc [hunk 1]
```diff
@@ -503,6 +507,22 @@ function ding_place2book_field_presave($entity_type, $entity, $field, $instance,
...
+        if (is_array($items) && isset($items[0]['event_id']) && isset($items[0]['event_maker_id'])) {
+          if ($items[0]['event_id'] == 1 && $items[0]['event_maker_id'] == 1) {
+            // This should not happen so we abort the saving off the node and alert the editor.
+
+            watchdog('ding_place2book', 'Original items: @original_items', array('@original_items' => print_r($settings , TRUE)), WATCHDOG_ERROR);
+            watchdog('ding_place2book', 'Synchronized items: @synchronized_items', array('@synchronized_items' => print_r($synchronized_items, TRUE)), WATCHDOG_ERROR);
+            watchdog('ding_place2book', 'Items to be saved: @items', array('@items' => print_r($items , TRUE)), WATCHDOG_ERROR);
+
+            drupal_set_message(t('Det lykkedes ikke at gemme din begivenhed. Du kan prøve igen senere, men hvis det stadig ikke lykkes,
+                er der desværre ingen anden mulighed end at oprette begivenheden på ny forfra.'), 'error');
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.fields.inc:520]: Please keep system texts in English.

File: modules/ding_place2book/ding_place2book.fields.inc [hunk 2]
```diff
@@ -503,6 +507,22 @@ function ding_place2book_field_presave($entity_type, $entity, $field, $instance,
...
           );
         }
+
+        if (is_array($items) && isset($items[0]['event_id']) && isset($items[0]['event_maker_id'])) {
+          if ($items[0]['event_id'] == 1 && $items[0]['event_maker_id'] == 1) {
+            // This should not happen so we abort the saving off the node and alert the editor.
+
+            watchdog('ding_place2book', 'Original items: @original_items', array('@original_items' => print_r($settings , TRUE)), WATCHDOG_ERROR);
+            watchdog('ding_place2book', 'Synchronized items: @synchronized_items', array('@synchronized_items' => print_r($synchronized_items, TRUE)), WATCHDOG_ERROR);
+            watchdog('ding_place2book', 'Items to be saved: @items', array('@items' => print_r($items , TRUE)), WATCHDOG_ERROR);
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.fields.inc:517]: Is there a reason to log this error in multiple entries? If not then consider combining messages and data into a single watchdog entry.

**Review by kasperg — CHANGES_REQUESTED** [review]
:+1: I appreciate the problems in dealing with an error that you cannot reproduce. This is as good as it gets.

I have a few comments for you to look into.

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 87a2c0b16670a7bb5761bed79f92b8acc6bb6764
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 87a2c0b16670a7bb5761bed79f92b8acc6bb6764
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 87a2c0b16670a7bb5761bed79f92b8acc6bb6764
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 87a2c0b16670a7bb5761bed79f92b8acc6bb6764
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 87a2c0b16670a7bb5761bed79f92b8acc6bb6764
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Review by kasperg — APPROVED** [review]
:+1: Thanks for the updates.

**Pull request merged**